ciphey/basemods/Decoders/brainfuck.py

```diff
@@ -0,0 +1,172 @@
+from typing import Optional, Dict, List, Tuple
+
+from ciphey.iface import Config, ParamSpec, T, U, Decoder, registry, WordList
+from ciphey.common import fix_case
+import time
+
+
+@registry.register
+class Brainfuck(Decoder[str, str]):
+    def decode(self, ctext: T) -> Optional[U]:
+        """
+        Takes a ciphertext and treats it as a Brainfuck program,
+        interpreting it and saving the output as a string to return.
+
+        Brainfuck is a very simple, Turing-complete esoteric language.
+        Below is a simplified interpreter that attempts to check whether a
+        given ciphertext is a brainfuck program that would output a string.
+
+        A program that can be "decoded" like this is one that:
+            * Does not require user input ("," instruction)
+            * Includes at least one putchar instruction (".")
+            * Does not contain anything but the main 7 instructions,
+                (excluding ",") and whitespace (tabs, spaces, and newlines)
+
+        Details:
+            * This implementation wraps the memory pointer for ">" and "<"
+            * It is time-limited to 60 seconds, to prevent hangups
+            * The program starts with 100 memory cells, chosen arbitrarily
+        """
+
+        result = ""
+        memory = [0] * 100
+        codeptr, memptr = 0, 0      # instruction pointer, stack pointer
+        timelimit = 60              # timeout, in seconds
```

Comment:
How likely is it for a false positive (a string that gets classed as brainfuck, but isn't) to fall through the cracks here? The timeout only really matters on that problem :)

Reply:
That's a good question. Honestly I figured you guys might have the better answer to that, since I'm not too familiar with the specifics of Ciphey's search algorithm yet.

That being said, the string would have to be consisting only of a combination of brainfuck's eight operators, spaces, newlines, or tabs. So perhaps if the input is just whitespace, it could get classified here. In that case though it would simply loop through everything practically instantly as all whitespace is ignored.

Reply:
Sounds like it'd be fairly unlikely. It may be a good idea to let the user decide on the time it takes through the config file ( https://github.com/Ciphey/Ciphey/wiki/The-Settings-File ) . Or perhaps when @Cyclic3 introduces levels we can alter the timeout based on the level selection (since that's levels of quickness, iirc)